src/lib/code-sample-editor.ts

```diff
@@ -113,38 +132,50 @@ export class CodeSampleEditor extends LitElement {
     if (changedProperties.has('project')) {
       this._findProjectAndRegister();
     }
-    if (changedProperties.has('files')) {
+    if (changedProperties.has('files') || changedProperties.has('filename')) {
       this._currentFileIndex = 0;
+      if (this.filename && this.files) {
+        for (let i = 0; i < this.files.length; i++) {
+          const file = this.files[i];
+          if (file.name === this.filename) {
+            this._currentFileIndex = i;
+            break;
+          }
+        }
+      }
       // TODO(justinfagnani): whyyyy?
       if (this._tabBar) {
         await this._tabBar.updateComplete;
```

Comment:
btw, this await was necessary because of some strange issue in mwc-tab-bar, can you check if it's still necessary? Especially because it's bound in the template. I should have opened an issue on MWC...

Reply:
It is still required, the tabs don't render otherwise. I don't understand why either, but I'll look into it another time.